### PR TITLE
Gh 2

### PR DESCRIPTION
**Pull request by Ross-Alexandra**

### CODE REVIEW

File: sudoku/pattern.py [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+
+from sudoku import Puzzle
+
+
+class PatternResponse:
+    def __init__(self):
+
+        self._results = {}
+
+    def add_response(self, position: tuple, value: int, candidate_reduction=False):
```
Comment (Ross-Alexandra-Review) [inline sudoku/pattern.py]: Why would a response default any values..... every pattern should have this value, so this should probably be required.

File: sudoku/pattern.py [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+    UNSPECIFIED = 0
+    MATCHING = 1
+    NON_MATCHING = 2
+    FILLED = 3
+    UNFILLED = 4
+
+    def __init__(self, pattern, resolution, candidate_reduction=False):
+        """
+        Args:
+            pattern(list): A list of lists representing the
```
Comment (Ross-Alexandra-Review) [inline sudoku/pattern.py]: A list of list of ints.

**Review by Ross-Alexandra-Review — CHANGES_REQUESTED** [review]
More review should be done once these changes are submitted as this was a cursory look.

File: sudoku/pattern.py (hunk 1 above)
Comment (Ross-Alexandra) [reply sudoku/pattern.py]: Addressed in https://github.com/Ross-Alexandra/SudokuPlayer/pull/12/commits/d711b046a444ed30efdee3903ef0607dcd7c3d35

File: sudoku/pattern.py (hunk 2 above)
Comment (Ross-Alexandra) [reply sudoku/pattern.py]: Addressed in https://github.com/Ross-Alexandra/SudokuPlayer/pull/12/commits/d711b046a444ed30efdee3903ef0607dcd7c3d35

**Pull request closed without merging**